framework/src/base/MooseApp.C

```diff
@@ -562,20 +600,45 @@ MooseApp::runInputFile()
 
   _action_warehouse.executeAllActions();
 
-  if (isParamValid("mesh_only"))
+  if (getParam<std::string>("split_mesh") != "")
```

Comment:
Why not `isParamValid()`?

Reply:
That always returns true for parameters that have a default.  I suppose split_mesh doesn't need a default though - I'll take a look.